client/tinychain/ml/dnn.py

```diff
@@ -64,15 +197,26 @@ def train(self, inputs, cost):
                 A.append(A_l.copy())
                 Z.append(Z_l)
 
-            m = inputs.shape[0]
+            m = inputs.shape[0]   
             dA = cost(A[-1]).sum() / m
 
             updates = []
+            
+
+            l_m,l_v,t=optimizer['l_m'],optimizer['l_v'],optimizer['t']
+            
             for i in reversed(range(0, num_layers)):
-                dA, d_weights, d_bias = layers[i].gradients(A[i], dA, Z[i + 1])
-                update = layers[i].update(d_weights, d_bias)
+                
+                updated_values = optimizer.optimize(A[i],dA,Z[i+1],decay_rate_1=0.9,
```

Comment:
Did you mean to leave these hard-coded values? `decay_rate_1`, `decay_rate_2`, `epsilon`? If so they can be constants in the AdamOptimizer class